test/pytest/service-desktop/test_messages.py

```diff
@@ -1,203 +1,267 @@
-# Copyright (c) 2017-2021, Mudita Sp. z.o.o. All rights reserved.
+# Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
 # For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
-
 import pytest
-from harness.interface.defs import status
-
-MESSAGES_PAGE_SIZE = 4
-
-
-@pytest.mark.service_desktop_test
-@pytest.mark.usefixtures("phone_unlocked")
-def test_get_messages_without_pagination(harness):
-    limit = MESSAGES_PAGE_SIZE - 1
-    get_body = {"category": "message", "limit": limit, "offset": 0}
-    ret = harness.endpoint_request("messages", "get", get_body)
-    assert ret["status"] == status["OK"]
-
-    messages = ret["body"]["entries"]
-    messages_count = len(messages)
-    assert messages_count == limit
-    message_types = [1, 2, 4, 8, 16, 18, 255]
-    for message in messages:
-        assert type(message["contactID"]) == int
-        assert type(message["messageBody"]) == str
-        assert type(message["messageID"]) == int
-        assert type(message["messageType"]) == int
-        assert type(message["createdAt"]) == int
-        assert type(message["threadID"]) == int
-        assert message["messageType"] in message_types
+from harness.request import TransactionError
+from harness.api.messages import GetMessagesCount, GetMessagesWithOffsetAndLimit, GetMessageById, \
+    GetMessagesByThreadIdWithOffsetAndLimit, AddMessage, DeleteMessageById
+
+
+class MessagesTester:
+    def __init__(self, harness):
+        self.harness = harness
+
+    def get_messages_count(self):
```

Comment:
I have djavu - there were tests which had the very same functions last year?

Reply:
I guess you are talking about functions of messages API from harness tests repo.